lib/src/rules/prefer_const_literals_to_create_immutables.dart

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.prefer_const_literals_to_create_immutables;
+
+import 'package:analyzer/analyzer.dart';
+import 'package:analyzer/dart/ast/ast.dart' show AstVisitor;
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:analyzer/src/generated/resolver.dart';
+import 'package:linter/src/analyzer.dart';
+
+const desc =
+    'Prefer use const literals as parameter of constructors on @immutable classes.';
+
+const details = '''
+
+**PREFER** using `const` for instantiating list and map literal used as
+parameters in immutable class instantiations.
+
+**BAD:**
+```
+@immutable
+class A {
+  A(this.v);
+  final v;
+}
+
+A a1 = new A([1]);
+A a2 = new A({});
+```
+
+**GOOD:**
+```
+A a1 = new A(const [1]);
+A a2 = new A(const {});
+```
+
+''';
+
+/// The name of `meta` library, used to define analysis annotations.
+String _META_LIB_NAME = 'meta';
+
+/// The name of the top-level variable used to mark a immutable class.
+String _IMMUTABLE_VAR_NAME = 'immutable';
+
+bool _isImmutable(Element element) =>
```

Comment:
The class is also public and can take an optional `reason` as an argument. Do we want to cover those cases here as well?

Reply:
I created #869 to track handling of Immutable.

Reply:
Thanks.